### ISSUE TITLE
`@let` fails when declared and used in `ng-container`

### ISSUE THREAD

**Issue by cexbrayat** [angular/angular#56768]
### Which @angular/* package(s) are the source of the bug?

compiler-cli

### Is this a regression?

No

### Description

The following component throws a compilation error:

```ts
@Component({
  selector: 'app-root',
  standalone: true,
  template: `
    <ng-container>
      @let name = 'Angular';
      <h1>Hello from {{ name }}!</h1>
    </ng-container>
  `,
})
export class App {}
```

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-lrxg6d?description=An%20angular-cli%20project%20based%20on%20@angular/animations,%20@angular/common,%20@angular/compiler,%20@angular/core,%20@angular/forms,%20@angular/platform-browser,%20@angular/platform-browser-dynamic,%20@angular/router,%20core-js,%20rxjs,%20tslib%20and%20zone.js&file=package.json,src%2Fmain.ts&template=node&title=Angular%20Starter

### Please provide the exception or error you saw

```true
Application bundle generation failed. [2.815 seconds]

✘ [ERROR] Could not resolve [object Object] / undefined [plugin angular-compiler]

  This error came from the "onStart" callback registered here:

    node_modules/@angular-devkit/build-angular/node_modules/@angular/build/src/tools/esbuild/angular/compiler-plugin.js:109:18:
      109 │             build.onStart(async () => {
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.1.0-rc.0
Node: 18.20.3
Package Manager: npm 10.2.3
OS: linux x64

Angular: 18.1.0-next.4
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1801.0-rc.0
@angular-devkit/build-angular   18.1.0-rc.0
@angular-devkit/core            18.1.0-rc.0
@angular-devkit/schematics      18.1.0-rc.0
@angular/cli                    18.1.0-rc.0
@schematics/angular             18.1.0-rc.0
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.7
```

### Anything else?

I think the issue is in the type-check block generation

**Comment by JoostK** [comment]
This is likely being fixed in #56752

**Comment by crisbeto** [comment]
Yep correct, it'll be fixed by #56752.
